From 6fe2ba75d82b2ae43131f27af91266eb9406aa85 Mon Sep 17 00:00:00 2001
From: zqm64 <zqm64@outlook.com>
Date: Sat, 15 Aug 2020 22:43:55 +0800
Subject: [PATCH 075/110] init: Import Xiaomi changes for ulysse

---
 init/main.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/init/main.c b/init/main.c
index 214404464272..bde1a60e3508 100644
--- a/init/main.c
+++ b/init/main.c
@@ -2,6 +2,7 @@
  *  linux/init/main.c
  *
  *  Copyright (C) 1991, 1992  Linus Torvalds
+ *  Copyright (C) 2018 XiaoMi, Inc.
  *
  *  GK 2/5/95  -  Changed to support mounting root fs via NFS
  *  Added initrd & change_root: Werner Almesberger & Hans Lermen, Feb '96
@@ -478,11 +479,21 @@ static void __init mm_init(void)
 	kaiser_init();
 }
 
+#ifdef CONFIG_MACH_XIAOMI_ULYSSE
+int fpsensor=1;
+int tpsensor=1;
+#endif
+
 asmlinkage __visible void __init start_kernel(void)
 {
 	char *command_line;
 	char *after_dashes;
 
+#ifdef CONFIG_MACH_XIAOMI_ULYSSE
+	char *p=NULL;
+	char *tp=NULL;
+#endif
+
 	set_task_stack_end_magic(&init_task);
 	smp_setup_processor_id();
 	debug_objects_early_init();
@@ -514,6 +525,24 @@ asmlinkage __visible void __init start_kernel(void)
 	build_all_zonelists(NULL, NULL, false);
 	page_alloc_init();
 
+#ifdef CONFIG_MACH_XIAOMI_ULYSSE
+	p = NULL;
+	p= strstr(boot_command_line,"androidboot.fpsensor=fpc");
+	if(p) {
+		fpsensor = 1;
+	} else{
+		fpsensor = 2;
+	}
+
+	tp = NULL;
+	tp = strstr(boot_command_line,"shenchao");
+	if(tp) {
+		tpsensor = 1;
+	} else{
+		tpsensor = 2;
+	}
+#endif
+
 	pr_notice("Kernel command line: %s\n", boot_command_line);
 	/* parameters may set static keys */
 	jump_label_init();
-- 
2.20.1

