From d5d7c62ba027cf070dda62f99b54dae9a260a880 Mon Sep 17 00:00:00 2001
From: zqm64 <zqm64@outlook.com>
Date: Sat, 18 Jul 2020 17:39:05 +0800
Subject: [PATCH 078/110] gpiolib: Import Xiaomi changes for ulysse

---
 drivers/gpio/gpiolib.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/gpio/gpiolib.c b/drivers/gpio/gpiolib.c
index 73d02f6089d5..54f45e3906da 100644
--- a/drivers/gpio/gpiolib.c
+++ b/drivers/gpio/gpiolib.c
@@ -2234,6 +2234,16 @@ static int _gpiod_direction_output_raw(struct gpio_desc *desc, int value)
 	struct gpio_chip *gc = desc->gdev->chip;
 	int ret;
 
+#ifdef CONFIG_MACH_XIAOMI_ULYSSE
+	if (test_bit(FLAG_USED_AS_IRQ, &desc->flags)) {
+		gpiod_err(desc,
+			  "%s: tried to set a GPIO tied to an IRQ as output\n",
+			  __func__);
+		if(desc_to_gpio(desc) != 65) {
+			return -EIO;
+		}
+	}
+#else
 	/* GPIOs used for IRQs shall not be set as output */
 	if (test_bit(FLAG_USED_AS_IRQ, &desc->flags)) {
 		gpiod_err(desc,
@@ -2241,6 +2251,7 @@ static int _gpiod_direction_output_raw(struct gpio_desc *desc, int value)
 			  __func__);
 		return -EIO;
 	}
+#endif
 
 	if (test_bit(FLAG_OPEN_DRAIN, &desc->flags)) {
 		/* First see if we can enable open drain in hardware */
-- 
2.20.1

