From 9c0da8454d1c84c1f95f3453db9fc26923320ee6 Mon Sep 17 00:00:00 2001
From: zqm64 <zqm64@outlook.com>
Date: Sun, 6 Sep 2020 04:09:57 +0800
Subject: [PATCH 4/5] clk: msm: gcc: Adapt GhiffaryR OC v3 changes for msm8917

---
 drivers/clk/msm/clock-gcc-8952.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/clk/msm/clock-gcc-8952.c b/drivers/clk/msm/clock-gcc-8952.c
index 123f0c245648..0b498bd80552 100644
--- a/drivers/clk/msm/clock-gcc-8952.c
+++ b/drivers/clk/msm/clock-gcc-8952.c
@@ -213,6 +213,9 @@ static struct pll_freq_tbl apcs_c0_pll_freq[] = {
 	F_APCS_PLL(1363200000,  71, 0x0, 0x1, 0x0, 0x0, 0x0),
 	F_APCS_PLL(1459200000,  76, 0x0, 0x1, 0x0, 0x0, 0x0),
 	F_APCS_PLL(1574400000,  82, 0x0, 0x1, 0x0, 0x0, 0x0),
+	F_APCS_PLL(1651200000,  86, 0x0, 0x1, 0x0, 0x0, 0x0),
+	F_APCS_PLL(1708800000,  89, 0x0, 0x1, 0x0, 0x0, 0x0),
+	F_APCS_PLL(1804800000,  94, 0x0, 0x1, 0x0, 0x0, 0x0),
 };
 
 static struct pll_clk a53ss_c0_pll = {
@@ -866,6 +869,8 @@ static struct clk_freq_tbl ftbl_gcc_oxili_gfx3d_clk_8917[] = {
 	F_SLEW( 523200000, 1046400000,	  gpll3,	1,	0,	0),
 	F_SLEW( 550000000, 1100000000,	  gpll3,	1,	0,	0),
 	F_SLEW( 598000000, 1196000000,	  gpll3,	1,	0,	0),
+	F_SLEW( 600000000, 1200000000,	  gpll3,	1,	0,	0),
+	F_SLEW( 620000000, 1240000000,	  gpll3,	1,	0,	0),
 	F_END
 };
 
@@ -888,6 +893,8 @@ static struct clk_freq_tbl ftbl_gcc_oxili_gfx3d_clk_8917_650MHz[] = {
 	F_SLEW( 523200000, 1046400000,	  gpll3,	1,	0,	0),
 	F_SLEW( 550000000, 1100000000,	  gpll3,	1,	0,	0),
 	F_SLEW( 598000000, 1196000000,	  gpll3,	1,	0,	0),
+	F_SLEW( 600000000, 1200000000,	  gpll3,	1,	0,	0),
+	F_SLEW( 620000000, 1240000000,	  gpll3,	1,	0,	0),
 	F_SLEW( 650000000, 1300000000,	  gpll3,	1,	0,	0),
 	F_END
 };
@@ -4283,13 +4290,13 @@ static void override_for_8917(int speed_bin)
 
 	if (speed_bin) {
 		OVERRIDE_FMAX5(gfx3d,
-			LOWER, 270000000, LOW, 400000000, NOMINAL, 484800000,
-			NOM_PLUS, 523200000, HIGH, 650000000);
+			LOWER, 240000000, LOW, 375000000, NOMINAL, 450000000,
+			NOM_PLUS, 600000000, HIGH, 650000000);
 		OVERRIDE_FTABLE(gfx3d, ftbl_gcc_oxili_gfx3d_clk, 8917_650MHz);
 	} else {
 		OVERRIDE_FMAX5(gfx3d,
-			LOWER, 270000000, LOW, 400000000, NOMINAL, 484800000,
-			NOM_PLUS, 523200000, HIGH, 598000000);
+			LOWER, 240000000, LOW, 375000000, NOMINAL, 450000000,
+			NOM_PLUS, 600000000, HIGH, 620000000);
 		OVERRIDE_FTABLE(gfx3d, ftbl_gcc_oxili_gfx3d_clk, 8917);
 	}
 
-- 
2.20.1

